Content.Server/Payload/EntitySystems/PayloadSystem.cs

```diff
@@ -0,0 +1,151 @@
+        SubscribeLocalEvent<PayloadCaseComponent, EntRemovedFromContainerMessage>(OnEntityRemoved);
+        SubscribeLocalEvent<ChemicalPayloadComponent, TriggerEvent>(HandleChemichalPayloadTrigger);
+    }
+
+    private void OnCaseTriggered(EntityUid uid, PayloadCaseComponent component, TriggerEvent args)
+    {
+        if (!TryComp(uid, out ContainerManagerComponent? contMan))
+            return;
+
+        // Pass trigger event onto all contained payloads. Payload capacity configurable by construction graphs.
+        foreach (var container in contMan.Containers.Values)
+        {
+            foreach (var entity in container.ContainedEntities)
+            {
+                if (_tagSystem.HasTag(entity, "Payload"))
+                    RaiseLocalEvent(entity, args, false);
+            }
+        }
+    }
+
+    private void OnTriggerTriggered(EntityUid uid, PayloadTriggerComponent component, TriggerEvent args)
+    {
+        if (!component.Active)
+            return;
+
+        if (Transform(uid).ParentUid is not { Valid: true } parent)
+            return;
+
+        // Ensure we don't enter a trigger-loop
+        DebugTools.Assert(!_tagSystem.HasTag(uid, "Payload"));
+
+        RaiseLocalEvent(parent, args, false);
+    }
+
+    private void OnEntityInserted(EntityUid uid, PayloadCaseComponent _, EntInsertedIntoContainerMessage args)
+    {
+        if (!TryComp(args.Entity, out PayloadTriggerComponent? trigger))
+            return;
+
+        trigger.Active = true;
+
+        if (trigger.Components == null)
+            return;
+
+        // ANY payload trigger that gets inserted can grant components. It is up to the construction graphs to determine trigger capacity.
+        foreach (var (name, data) in trigger.Components)
+        {
+            if (!_componentFactory.TryGetRegistration(name, out var registration))
+                continue;
+
+            if (HasComp(uid, registration.Type))
+                continue;
+
+            if (_componentFactory.GetComponent(registration.Type) is not Component component)
+                continue;
+
+            component.Owner = uid;
+
+            if (_serializationManager.Copy(data, component, null) is Component copied)
+                EntityManager.AddComponent(uid, copied);
+
+            trigger.GrantedComponents.Add(registration.Type);
+        }
+    }
+
+    private void OnEntityRemoved(EntityUid uid, PayloadCaseComponent component, EntRemovedFromContainerMessage args)
+    {
+        if (!TryComp(args.Entity, out PayloadTriggerComponent? trigger))
+            return;
+
+        trigger.Active = false;
+
+        foreach (var type in trigger.GrantedComponents)
+        {
+            EntityManager.RemoveComponent(uid, type);
+        }
+
+        trigger.GrantedComponents.Clear();
+    }
+
+    private void HandleChemichalPayloadTrigger(EntityUid uid, ChemicalPayloadComponent component, TriggerEvent args)
```

Comment:
```suggestion
    private void HandleChemicalPayloadTrigger(EntityUid uid, ChemicalPayloadComponent component, TriggerEvent args)
```

Reply:
Fun fact, I am for whatever reason incapable of NOT making that spelling mistake. I think every time I've typed out the word chemical I've had to go back and correct it. Imma do a ctrl+f and look for more instances, because there probably are